src/model/Course.java

```diff
@@ -12,36 +12,949 @@
+	 * others for the same course
 	 */
-	Date creationDate;
-	
+	private LocalDateTime creationDate;
+
 	/**
 	 * The courses that should be completed before taking this one
 	 */
-	ArrayList<String> prerequisites;
-	
+	private ArrayList<String> prerequisites;
+
 	/**
 	 * Specifies the course language
 	 */
-	Language courseLang;
+	private Language courseLang;
+
+	/**
+	 * Specifies the semester(s) the course will be given
+	 */
+	private SemesterOptions semester;
+
+	/**
+	 * Shows the local credit for the course and calculated by the program itself
+	 * according to the pseudo code below ; localCredit = round(theoreticalHour +
+	 * (0.5 * labHour))
+	 * 
+	 * @see #calculateLocalCredit()
+	 */
+	private int localCredit;
+
+	/**
+	 * Shows the ECTS equivalent value of the course This value is calculated by the
+	 * program itself according the pseudo code below ; ECTS = round(sumAll(workload
+	 * of each element in workloadTable) / 30)
+	 * 
+	 * @see #calculateECTS()
+	 *
+	 */
+	private int ects;
+
+	/**
+	 * Theoretical hours per week for the course.
+	 */
+	private int theoreticalHour;
+
+	/**
+	 * Lab/Application hours per week for the course.
+	 */
+	private int labHour;
+
+	/**
+	 * Course type
+	 */
+	private CourseType type;
+
+	/**
+	 * Coordinator of the course. There can only be 1 coordinator and he must be a
+	 * teacher
+	 */
+	private Lecturer courseCordinator;
+
+	/**
+	 * Lecturers who give this course. There can be more then one lecturer
+	 * 
+	 * @see #addLecturer(Lecturer)
+	 * @see #removeLecturer(Lecturer)
+	 */
+	private ArrayList<Lecturer> courseLecturers;
+
+	/**
+	 * Assistants of this course. There can be more then one assistant.
+	 * 
+	 * @see #addAssistant(Lecturer)
+	 * @see #removeAssistant(Lecturer)
+	 */
+	private ArrayList<Lecturer> assistants;
+
+	/**
+	 * Course objective summary.
+	 */
+	private String courseObjective;
+
+	/**
+	 * Learning outcomes as a String ArrayList. Order of the array is relevant
+	 * because {@link EvaluationCriteria} field is using the order of the array the
+	 * identify different learning outcomes.
+	 * 
+	 * @see #addCriteria(EvaluationCriteria)
+	 */
+	private ArrayList<String> learningOutcomes;
+
+	/**
+	 * Course Description summary.
+	 */
+	private String courseDescription;
+
+	/**
+	 * One of the categories from {@link CourseCategory} enumerator.
+	 * 
+	 * @see CourseCategory
+	 */
+	private CourseCategory courseCategory;
+
+	/**
+	 * Course Schedule with 16 weeks (16 object). Weeks are created at the
+	 * construction state of the object. User will not be able to delete a week. But
+	 * field(s) of a {@link WeeklySubject} may be changed to accommodate the
+	 * requirements.
+	 * 
+	 * @see #changeSchedule(int, String, String)
+	 */
+	private ArrayList<WeeklySubject> courseSchedule;
+
+	/**
+	 * General course textbook.
+	 */
+	private CourseBook courseTextBook;
+
+	/**
+	 * Suggested readings for this course
+	 */
+	private ArrayList<CourseBook> suggestedReading;
```

Comment:
suggested reading aslında o hafta okunacak kitabın ilgili chapter'i oluyor, yani burada bir dersin neredeyse her hafta aynı kitabı kullandığını düşünebilir miyiz?

Reply:
Suggested reading olarak kastettiğim bölüm kurs müfredatlarında "Course Notes/ Textbook"un altında belirttiği dersle alakalı diğer kitap önerilerini içermektedir. Belirli hafta için okunacak kitaplar veya bölümün önerildiği yer ise "WeeklySubject" class'ının "relatedMaterial" alanı olarak tanımlamıştım. Her hafta için müfredatta önerilen bölüm haftalık konu objelerinin içindedir. Fakat açıklamanın anlaşılabilirliği ve bu konuya açıklık getirmesi için genişletilmesi gerekiyor bence de. Göz atıp, açıklamayı daha detaylı hale getireceğim. Yorumun için teşekkürler.